### PR TITLE
Cookie Authentication

### PR DESCRIPTION
**Pull request by wintondeshong**
Fixes issue #34 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+using AndcultureCode.CSharp.Core.Extensions;
+using Microsoft.Extensions.Logging;
+using AndcultureCode.GB.Presentation.Web.Middleware.Authentication;
+using AndcultureCode.CSharp.Core.Interfaces;
+using AndcultureCode.GB.Business.Core.Interfaces.Conductors.Domain.Users;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Authorization;
+using AndcultureCode.CSharp.Business.Core.Models.Configuration;
+
+namespace LMS.Presentation.Web.Controllers.Api.V1.UserLogins
```
Comment (brandongregoryscott) [inline dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs:25]: Namespace here looks off

File: dotnet/api/Tests/Testing/Constants/Trait.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+namespace Testing.Constants
```
Comment (brandongregoryscott) [inline dotnet/api/Tests/Testing/Constants/Trait.cs:1]: Should this namespace be `AndcultureCode.GB.Api.Tests.Testing.Constants`?

File: dotnet/api/Tests/Testing/Constants/TraitCategory.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+namespace Testing.Constants
```
Comment (brandongregoryscott) [inline dotnet/api/Tests/Testing/Constants/TraitCategory.cs:1]: Same here

File: dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs (hunk 1 above)
Comment (wintondeshong) [reply dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs:25]: Bah!

File: dotnet/api/Presentation/Web/Controllers/Api/V1/Roles/RolesController.cs [hunk 4]
```diff
@@ -52,22 +51,7 @@ IRepositoryConductor<Role> repositoryConductor
...
-
-            var role = readResult.ResultObject;
-            if (role == null)
-            {
-                return NotFound<RoleDto>();
-            }
-
-            return Ok(_mapper.Map<RoleDto>(role), null);
-        }
+        public IActionResult Get(long id) => this.GetDefault<RolesController, Role, RoleDto>(id);
```
Comment (Stefanie899) [inline dotnet/api/Presentation/Web/Controllers/Api/V1/Roles/RolesController.cs:54]: Love how clean this is.

File: dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            var userLoginResult = _conductor.FindAll(e => e.Ip == IpAddress && e.UserName == userName);
+            var userLogin = userLoginResult.ResultObject?.OrderByDescending(e => e.CreatedOn)?.FirstOrDefault();
+
+            // If last UserLogin is null or was successful create a new failed one.
+            if (userLogin == null || userLogin.IsSuccessful)
+            {
+                var createResult = CreateUserLogin(userName, isSuccessful: false);
+                if (createResult.HasErrorsOrResultIsNull())
+                {
+                    errors = createResult.Errors;
```
Comment (Stefanie899) [inline dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs:143]: Should this be returning an `InternalError` if it fails to create the UserLogin?

Comment (wintondeshong) [reply dotnet/api/Presentation/Web/Controllers/Api/V1/UserLogins/UserLoginsController.cs:143]: Yeah, that's a good point! Updating it

**Pull request merged**